### DISCUSSION TITLE
Remove the old `SplitChunksPlugin` from the codebase

### DISCUSSION THREAD

**Discussion by hyf0** [web-infra-dev/rspack#4168] [Core]
After implementing the new `SplitChunksPlugin` and the tests being passed, Rspack enable the new `SplitChunksPlugin` by default. It brings better performance and stability.And there's one issue remains unresolved, which is https://github.com/web-infra-dev/rspack/issues/3120.

But now https://github.com/web-infra-dev/rspack/issues/3120 is solved, which leave us no reasons to keep the old `SplitChunksPlugin` anymore.

So I think it's time to remove the old `SplitChunksPlugin` from the codebase, it will reduce the binary size of rspack and complexity of the codebase.

I don't consider this a breaking change, since the new `SplitChunksPlugin` is enable by default and we are removing a option in `experiments`.

**Comment by hardfist** [comment]
it would be better it's released in 0.4.0 incase different behavior caused by new Implementation and old Implementation inconsistencies, people may rely on some css bug caused by old behavior

**Reply by hyf0** [reply]
Agree. I will do the remove step at https://github.com/web-infra-dev/rspack/pull/4041.
